vowpalwabbit/options.h

```diff
@@ -88,25 +91,36 @@ struct typed_option : base_option
   typed_option& value(T value)
   {
     m_value = std::make_shared<T>(value);
+    if (m_location != nullptr) { *m_location = value; }
+    value_set_callback(value);
     return *this;
   }
 
   T value() const { return m_value ? *m_value : T(); }

```

Comment:
semi related since we're here: isn't this code wrong? shouldn't it put the default value after calling into T()?

Reply:
`T()` is a fallback for if there is no value provided, it's the default for `T` but necessarily a default value in the sense of the option

Reply:
oh yea, typed option doesn't even have knowledge of default value